spec/arel/middleware_spec.rb

```diff
@@ -423,9 +425,10 @@ def self.call(_arel, next_middleware)
     after_sql = []
 
     logger = lambda do |next_arel, next_middleware, context|
+      result = next_middleware.call(next_arel)
       after_sql << next_arel.to_sql
       before_sql << context[:original_sql]
-      next_middleware.call(next_arel)
+      result
```

Comment:
Interesting that you got you an out of order array of SQLs without this change.

Reply:
it is even more interesting why the old code works with arel 9